### PR TITLE
JSON Schema v0.2: Hash the UACs

### PR DESCRIPTION
**Pull request by LeoKHoward**
# Motivation and Context
We should NEVER emit a plain text UAC, for security reasons.
We should emit a SHA-256 hash of the UAC instead, per the JSON Schema spec.

# What has changed
DTO field changed to uacHash
New HashHelper to convert String and/or Bytes to Hash
Tests

# How to test?
Maven clean install
Run ATs on branch of same name

# Links
https://github.com/ONSdigital/ssdc-shared-events/blob/main/event_dictionary/v0.2_RELEASE/dictionary.md#payload_oneOf_i8_uacUpdate_uacHash
https://trello.com/c/jB50vAb8

### CODE REVIEW

**Comment by LeoKHoward** [comment]
[JSON Schema v0.2 RELEASE : UACs should be hashed and don't emit plain text UAC (5)](https://trello.com/c/jB50vAb8/2674-json-schema-v02-release-uacs-should-be-hashed-and-dont-emit-plain-text-uac-5)

File: src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/TelephoneCaptureReceiverIT.java [hunk 1]
```diff
@@ -18,19 +18,15 @@
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 import uk.gov.ons.ssdc.caseprocessor.client.UacQidServiceClient;
-import uk.gov.ons.ssdc.caseprocessor.model.dto.EventDTO;
-import uk.gov.ons.ssdc.caseprocessor.model.dto.EventHeaderDTO;
-import uk.gov.ons.ssdc.caseprocessor.model.dto.PayloadDTO;
-import uk.gov.ons.ssdc.caseprocessor.model.dto.TelephoneCaptureDTO;
-import uk.gov.ons.ssdc.caseprocessor.model.dto.UacQidDTO;
-import uk.gov.ons.ssdc.caseprocessor.model.dto.UacUpdateDTO;
+import uk.gov.ons.ssdc.caseprocessor.model.dto.*;
```
Comment (NickGrantONS) [inline src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/TelephoneCaptureReceiverIT.java:21]: you need to use the proper Google coding standards, not any plugin

**Review by NickGrantONS — APPROVED** [review]

File: src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/TelephoneCaptureReceiverIT.java (hunk 1 above)
Comment (LeoKHoward) [reply src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/TelephoneCaptureReceiverIT.java:21]: Yes this is what I use: https://github.com/google/google-java-format

Plugin developed by Google...the method of adding an XML file to the Code Style Scheme seems to be the outdated approach now

Comment (NickGrantONS) [reply src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/TelephoneCaptureReceiverIT.java:22]: ok, team standards then... you are the only person who changes all the imports

**Review by NickGrantONS — CHANGES_REQUESTED** [review]
Need to comply with _team_ coding standards. You can't keep changing tons of code because you have non-standard settings. The standard used by the team is to use XML IDE setup, so that imports don't get all messed up

**Review by NickGrantONS — APPROVED** [review]

**Review by ryangrundy7 — APPROVED** [review]

**Pull request merged**